### PR TITLE
fix layering violations in component updater and add missing dep

### PR DESCRIPTION
**Pull request by bridiver**
## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [x] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [x] macOS
  - [ ] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [x] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [x] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protection-Mode
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: browser/extensions/brave_extension_install_prompt.cc [hunk 1]
```diff
@@ -5,12 +5,14 @@
...
 
-#include "brave/browser/extensions/brave_extension_provider.h"
+#include "brave/browser/brave_browser_process_impl.h"
+#include "brave/components/brave_component_updater/browser/extension_whitelist_service.h"
 #include "brave/grit/brave_generated_resources.h"
 #include "ui/base/l10n/l10n_util.h"
 
 base::string16 BravePrompt::GetDialogTitle() const {
-  if (!extensions::BraveExtensionProvider::IsVetted(extension())) {
+  if (g_brave_browser_process->extension_whitelist_service()->IsVetted(
```
Comment (mkarolin) [inline browser/extensions/brave_extension_install_prompt.cc]: `!` got lost.

Comment (bridiver) [reply browser/extensions/brave_extension_install_prompt.cc]: fixed

File: browser/extensions/brave_extension_provider.h [hunk 2]
```diff
@@ -21,8 +21,7 @@ class BraveExtensionProvider : public ManagementPolicy::Provider {
                    base::string16* error) const override;
   bool MustRemainInstalled(const Extension* extension,
                            base::string16* error) const override;
-  static bool IsVetted(const extensions::Extension* extension);
-  static bool IsVetted(const std::string id);
+
```
Comment (mkarolin) [inline browser/extensions/brave_extension_provider.h]: Nit: extra line?

File: components/brave_component_updater/browser/extension_whitelist_service.cc [hunk 3]
```diff
@@ -38,6 +43,23 @@ bool ExtensionWhitelistService::IsBlacklisted(
   return extension_whitelist_client_->isBlacklisted(extension_id.c_str());
 }
 
+bool ExtensionWhitelistService::IsVetted(const Extension* extension) const {
+  // This is a hardcoded list of vetted extensions, mostly
```
Comment (mkarolin) [inline components/brave_component_updater/browser/extension_whitelist_service.cc]: This comment is probably better suited for `whitelist.cc` now.

File: browser/extensions/brave_extension_provider.h (hunk 2 above)
Comment (bridiver) [reply browser/extensions/brave_extension_provider.h]: a blank line before `private:` is required by lint

**Pull request merged**